src/profile-logic/profile-data.js

```diff
@@ -1027,6 +1027,53 @@ export function filterThreadToSearchString(
   );
 }
 
+export function filterThreadByTab(
+  thread: Thread,
+  relevantPages: Set<InnerWindowID>
+): Thread {
+  return timeCode('filterThreadByTab', () => {
+    if (relevantPages.size === 0) {
+      return thread;
+    }
+
+    const { frameTable, stackTable } = thread;
+
+    const frameMatchesFilterCache = new Map();
+    function frameMatchesFilter(frame) {
+      let result = frameMatchesFilterCache.get(frame);
+      if (result === undefined) {
+        const innerWindowID = frameTable.innerWindowID[frame];
+        result = innerWindowID && relevantPages.has(innerWindowID);
```

Comment:
I find the `&&` operators extremely hard to reason with. Would you mind separating out the condition and the assignment? I usually find the ternary is much easier to read.

Reply:
Also, please add some strict equality checking here. The implicit is dangerous, especially as there are no comments here explaining things. I THINK innerWindowID will be 0 when it can't find one.